src/pd/client.rs

```diff
@@ -210,10 +210,15 @@ impl PdClient for RpcClient {
         })?;
         check_resp_header(resp.get_header())?;
 
-        Ok(resp.take_store())
+        let store = resp.take_store();
+        if store.get_state() != metapb::StoreState::Tombstone {
+            Ok(store)
+        } else {
+            Err(Error::StoreTombstone(format!("{:?}", store)))
```

Comment:
Why return error when the store is `Tombstone`?

Reply:
Tombstone stores are no longer belongs to the cluster, other TiKVs should ignore such stores. See more pingcap/tidb#9269

Reply:
seem we don't need to use a string for the StoreTombstone. maybe integer is enough.

Reply:
PD may return StoreTombston error too, but it does not contain a store id, so I choose string for simplicity.